app/assets/javascripts/discourse/app/models/user.js

```diff
@@ -337,16 +337,18 @@ const User = RestModel.extend({
       return [];
     }
 
-    return Site.current().categoriesList.filter((category) => {
-      if (
-        this.siteSettings.suppress_uncategorized_badge &&
-        category.isUncategorizedCategory
-      ) {
-        return false;
-      }
+    return Site.current()
+      .categoriesList.filter((category) => {
```

Comment:
Another minor thought, maybe we could add a `sortedCategoriesList` function to `Site.current()`. No need to do it now, just thinking out loud :+1:

Reply:
Actually `categoriesList` is sorted depending on a site setting.

https://github.com/discourse/discourse/blob/4657110c35cd5684d4554bef1eb4a1287d2d4aab/app/assets/javascripts/discourse/app/models/site.js#L79-L84

For sidebar right now, we want it to be sorted by name regardless so I kept the sorting logic out of `Site`.

Reply:
Ah yes fair enough, thanks for pointing that out :+1: